android/src/main/java/com/swmansion/rnscreens/ScreenStack.kt

```diff
@@ -226,16 +228,28 @@ class ScreenStack(
             needsDrawReordering(newTop, stackAnimation) &&
             visibleBottom == null
         ) {
-            // When using an open animation in which two screens overlap (eg. fade_from_bottom or
-            // slide_from_bottom), we want to draw the previous screen under the new one,
-            // which is apparently not the default option. Android always draws the disappearing view
+            // When using an open animation in which screens overlap (eg. fade_from_bottom or
+            // slide_from_bottom), we want to draw the previous screens under the new one,
+            // which is apparently not the default option. Android always draws the disappearing views
             // on top of the appearing one. We then reverse the order of the views so the new screen
-            // appears on top of the previous one. You can read more about in the comment
+            // appears on top of the previous ones. You can read more about in the comment
             // for the code we use to change that behavior:
             // https://github.com/airbnb/native-navigation/blob/9cf50bf9b751b40778f473f3b19fcfe2c4d40599/lib/android/src/main/java/com/airbnb/android/react/navigation/ScreenCoordinatorLayout.java#L18
             // Note: This should not be set in case there is only a single screen in stack or animation `none` is used.
             // Atm needsDrawReordering implementation guards that assuming that first screen on stack uses `NONE` animation.
-            childDrawingOrderStrategy = SwapLastTwo()
+
+            // Count how many screens are disappearing
+            // (all transparent screens beneath newTop + one screen under them)
+            val disappearingCount =
+                notDismissedWrappers
+                    .drop(1)
+                    .takeWhile { it.screen.isTransparent() }
+                    .count() + 1
```

Comment:
I'm reading the code once again now & my initial impression is that it won't work in general case. I need to come up with counter example -- will try to do that next week.

Reply:
The problem is that `notDismissedWrappers` sequence is computed basing on `screenWrappers` - these represent **current** react view state. We don't have guarantees that this is consistent with **currently HostTree state** represented by `stack` list. 

I think we need to find some time in upcoming weeks & design this algorithm properly from the ground up.

Reply:
Notation: 

* uppercase letters, e.g. A, B - **opaque** screens,
* lowercase letters, e.g. a, b - **translucent** screens.

`stack`: `Abc`
`screenWrappers`: `AD`

This case will have issue, where the count of dismissed screens will be 0 -> therefore we won't change drawing order, while we should.

One thing to note that current solution might be good enough to work in simple stack operations (push, pop, replace). The one I've just described is not a stack operation. 

My aspiration is to make the screen stack work with arbitrary state changes, but I guess we can postpone this plant until we have more robust update algorithm.